Style Guide for Help Text

knoll-alex opened the pull request:
Hello! I created a Style Guide for future iterations of the user-facing help text in LORIS. This is a set of guidelines to follow for consistency.

#### Testing instructions

1. Review for content. Comment as necessary. 
2. Ensure image is appearing.

Thanks for welcoming me on the LORIS team for a brief time!

christinerogers commented on docs/HelpStyleGuide.md:
```diff
@@ -0,0 +1,66 @@
+    * i.e. In the **Imaging Browser** module
+    * i.e. Click the **Upload** tab
+  * You can also bold when defining features in a list, to separate the term being defined from the definition itself, i.e. when describing columns in a table:
+     **Column heading 1**: description
```
These aren't rendering on their own lines.  Take a look, @knoll-alex ?

knoll-alex replied:
thanks! this should be fixed.

PapillonMcGill commented on docs/HelpStyleGuide.md:
```diff
@@ -0,0 +1,66 @@
+  * i.e. clicking a Download button to download something
+  * i.e. navigate pages of a table using the arrows provided
+* Use [Plain Language](https://www.plainlanguage.gov/resources/articles/elements-of-plain-language/)
+  * As a guideline, refer to [this list](https://www.plainlanguage.gov/guidelines/words/use-simple-words-phrases/) of words to avoid, and alternate suggestions 
```
I don't like that we refer to foreign government rules. Not only theses can change without us noticing but they may not be the best in our situation.

christinerogers replied:
@PapillonMcGill I think we could still include these links for context(inspiration), without saying that we are strictly following them -- would you be ok with that?

PapillonMcGill commented on docs/HelpStyleGuide.md:
```diff
@@ -0,0 +1,66 @@
+     
+## Tone
+
+Yes, LORIS is used for high-level research and yes, the people using LORIS are very smart and yes, we want people in the scientific community to take LORIS seriously. At the same time, the easier any user finds a text to read, the more they’ll retain and use the information. And a friendly tone will help them understand that LORIS is built and maintained by humans. There’s a lot of research that  ...
```
replace lines 48-50 with 
`LORIS is used by scientific community worldwide, this include countries where English is a second or even third language. Thus, it is important that LORIS documentation (instruction, help text, ...) be written in a way comprehensible by people with limited knowledge and understanding of English language and expression.`
or something more similar.

christinerogers replied:
great suggestions @PapillonMcGill,  and this could probably be shortened as you suggest.

driusan commented on docs/HelpStyleGuide.md:
```diff
@@ -0,0 +1,66 @@
+     
+## Tone
+
+Yes, LORIS is used for high-level research and yes, the people using LORIS are very smart and yes, we want people in the scientific community to take LORIS seriously. At the same time, the easier any user finds a text to read, the more they’ll retain and use the information. And a friendly tone will help them understand that LORIS is built and maintained by humans. There’s a lot of research that  ...
```
Do you have any references for "a lot of research" (both out of curiousity, and in case someone asks in the future) that could be referenced here?

knoll-alex replied:
Hi @driusan - I removed this line (along with the rest of that paragraph), but still happy to provide some context to this claim. I added an article at the end that talks about Plain Language, and cites a usability study that was conducted in scientific fields. I hope this will suffice!

knoll-alex commented:
Thank you @PapillonMcGill for your review and suggestions! I have updated the content to better reflect the tone you're looking for. I wasn't sure if you wanted me to remove line 50 as well - let me know and I can remove it if need be. I removed links to the government website, and instead added an article at the end which references research-based information on Plain Language instead.

christinerogers commented:
Hi @knoll-alex This week the team is rebasing all of our PRs open on `major` to `master` -- 
If this is doable for you, please take a moment to hop this onto `master`, and if not, don't hesitate to let me know.  Thanks!

PapillonMcGill commented:
Yes please, could you remove line 50?

christinerogers commented on docs/HelpStyleGuide.md:
```diff
@@ -0,0 +1,68 @@
+     
+## Tone
+
+LORIS is used by scientific community worldwide, including countries where English is a second, or even third language. Therefore, it is important that LORIS documentation (help text, setup guides, etc.) be written in a way that it comprehensible by people with limited knowledge and understanding of English language and expression.
```
To shorten, and fix 2 grammar issues: 
```suggestion
LORIS is used by scientific community members worldwide, which means it's possible that most users are non-native English speakers. Writing our Help text or any LORIS documentation (setup guides, etc.) in a way that it is quickly comprehensible by people with many different levels of English language and expression.
```

PapillonMcGill replied:
`... in a way that is quickly ...`

I would just do a little change to the last sentence.

christinerogers replied:
Ok thanks - i updated my suggestion to include your change @PapillonMcGill

christinerogers commented on docs/HelpStyleGuide.md:
```diff
@@ -0,0 +1,68 @@
+
+LORIS is used by scientific community worldwide, including countries where English is a second, or even third language. Therefore, it is important that LORIS documentation (help text, setup guides, etc.) be written in a way that it comprehensible by people with limited knowledge and understanding of English language and expression.
+
+Please note, this is NOT about “dumbing things down”. It’s simply about using the most direct language and tone to get the information across to the user in the least amount of time, with the least amount of friction. The quicker a user understands something, the happier they are, the more trust they have with the software, and the less silly questions you can expect to get to the dev mailing lis ...
```
Paring this down just a little, after chatting with @PapillonMcGill 
```suggestion
This means using more direct language and tone to get information across to the user in the least amount of time, with the least amount of friction. The quicker a user understands something, the happier they are.  Easily understandable documentation also builds trust in the software, in addition to decreasing the amount of support questions we get.
```

johnsaigle commented on docs/HelpStyleGuide.md:
```diff
@@ -0,0 +1,68 @@
+
+## Markdown
+
+Formatting should follow markdown guidelines, with some exceptions. This is because LORIS’s current markdown template may not cover all possible markdown features (i.e. bullets). Help text should not include HTML tags for security reasons. When creating a markdown file, use only the following styles, or else talk to the LORIS technical team about expanding our Markdown features:
```
```suggestion
Formatting should follow markdown guidelines, with some exceptions. This is because LORIS’s current markdown template may not cover all possible markdown features (e.g. bullets). Help text cannot be formatted with HTML so it should not be included in help text. When creating a markdown file, use only the following styles, or else talk to the LORIS technical team about expanding our Markdown features:
```

Using "e.g." instead of "i.e." because there are many other Markdown features we don't support.

Also rewording the bit about HTML; it doesn't pose a security problem for us as the values are properly escaped. However it might make the text look funny so it's best to avoid it.

johnsaigle commented on docs/HelpStyleGuide.md:
```diff
@@ -0,0 +1,68 @@
+* Avoid using HTML line break (`<br>`) tags—simply use physical line breaks (spaces between lines) where necessary. 
+  * Using `<br>` might be necessary only in lists of items
+  
+  ## When to Use Emphasis
```
```suggestion
## When to Use Emphasis
```

I think this needs to be left-aligned in order to render properly as as heading.

johnsaigle commented on docs/HelpStyleGuide.md:
```diff
@@ -0,0 +1,68 @@
+  * DO call it a module or tab
+    * i.e. In the **Imaging Browser** module
+    * i.e. Click the **Upload** tab
+  * You can also use bold when defining features in a list, to separate the term being defined from the definition itself, i.e. when describing columns in a table:<br>
```
```suggestion
  * You can also use bold when defining features in a list, to separate the term being defined from the definition itself, e.g. when describing columns in a table:<br>
```

johnsaigle commented on docs/HelpStyleGuide.md:
```diff
@@ -0,0 +1,68 @@
+* Keep it concise. Avoid unnecessary information.
+  * Try not to be repetitive. 
+  * i.e. the Help text for the Imaging Browser module begins with the title Imaging Browser. You don’t need to start text with “In the Imaging Browser module...”, because the user already knows the module based on the title. Instead, keep it short and say “In this module…”
+* Assume the user is tech-savvy enough in the year 2020 to know about features that are intuitive or common across modern platforms, so you don’t need to repeat in every Help text
```
```suggestion
* Assume the user is tech-savvy enough to know about features that are intuitive or common across modern platforms, so you don’t need to repeat in every Help text
```

I see why this was added stylistically but I know that we'll forget to keep this current.

PapillonMcGill commented on docs/HelpStyleGuide.md:
```diff
@@ -0,0 +1,68 @@
+
+LORIS is used by scientific community members worldwide, which means it's possible that most users are non-native English speakers. Therefore, it is important that LORIS documentation (setup guides, etc.) be written in a way that is quickly comprehensible by people with many different levels of English language and expression.
+
+This means using more direct language and tone to get information across to the user in the least amount of time, with the least amount of friction. The quicker a user understands something, the happier they are.  Easily understandable documentation also builds trust in the software, in addition to decreasing the amount of support questions we get.
```
```suggestion
This means using more direct language and tone to get information across to the user in the least amount of time, with the least amount of friction. The quicker a user understands something, the happier they are. Easily understandable documentation also builds trust in the software, in addition to decreasing the amount of support questions we get.
```

PapillonMcGill commented on docs/HelpStyleGuide.md:
```diff
@@ -0,0 +1,68 @@
+     
+## Tone
+
+LORIS is used by scientific community members worldwide, which means it's possible that most users are non-native English speakers. Therefore, it is important that LORIS documentation (setup guides, etc.) be written in a way that is quickly comprehensible by people with many different levels of English language and expression.
```
```
... LORIS documentation (setup guides, etc.) ...
```
Why `help text` as been eliminated?  An enumeration with only 1 value before `etc`is not proper.

knoll-alex replied:
oops! added it back in

christinerogers replied:
my bad prolly

knoll-alex commented:
> Suggested some small changes. Largely looks good and I like the note about plain language!
> 
> In addition to my comments I think there is a bit of an overuse of "i.e." throughout the text and in most cases "e.g." would be more appropriate. It would be nice to fix but I won't block merging this PR because of it.

I've replaced all I.e. with e.g. - thank you for this suggestion. I prefer the way "I.e." looks but sometimes forget it really doesn't mean "for example" !

christinerogers commented:
Thanks for the last push to get this reviewed, tweaked and merged, guys !
@knoll-alex appreciate the extra edits
@PapillonMcGill @johnsaigle could you re-review?  Let's get this mergeable today. 🔥

johnsaigle commented:
The image doesn't appear to be rendering but I'm not sure if this is a bug with GitHub. I haven't tried to add in-line images before.

christinerogers commented:
> The image doesn't appear to be rendering but I'm not sure if this is a bug with GitHub. I haven't tried to add in-line images before.

It renders ok for me in the latest version here: https://github.com/aces/Loris/blob/d3b84eb22ba1ffd39e970aab21a697c394328afe/docs/HelpStyleGuide.md

Which version are you looking at, @johnsaigle ?

johnsaigle commented:
You're right. I was looking at the diff. I forgot GitHub has a two-step process for rendering docs

christinerogers commented:
@driusan looks pretty ready to merge.

driusan commented on docs/HelpStyleGuide.md:
```diff
@@ -0,0 +1,68 @@
+
+## Markdown
+
+Formatting should follow markdown guidelines, with some exceptions. This is because LORIS’s current markdown template may not cover all possible markdown features (e.g. bullets). Help text cannot be formatted with HTML so it should not be included in help text. When creating a markdown file, use only the following styles, or else talk to the LORIS technical team about expanding our Markdown featu ...
```
```suggestion
Formatting should follow markdown guidelines, with some exceptions. This is because LORIS’s current markdown renderer may not cover all possible markdown features (e.g. bullets). Help text cannot be formatted with HTML so it should not be included in help text. When creating a markdown file, use only the following styles, or else talk to the LORIS technical team about expanding our Markdown features:
```

This is more accurate

driusan commented on docs/HelpStyleGuide.md:
```diff
@@ -0,0 +1,68 @@
+  
+## When to Use Emphasis
+  
+Whether a user knows it or not, they will come to recognize styles that are consistent throughout the text. This way, their eyes can jump to a part of the text that addresses their issue, if they don’t need to read the whole thing. Follow these rules on using emphasis:
```
```suggestion
Whether a user knows it or not, they will come to recognize styles that are consistent throughout the text. This way, their eyes can jump to a part of the text that addresses their issue, and they don’t need to read the whole thing. Follow these rules on using emphasis:
```

The flow of the sentence is hard to follow because of the "," before "if".. changing it to "and" makes it easier for me to understand (or you could drop the comma, but then it would be a long sentence..)

driusan commented on docs/HelpStyleGuide.md:
```diff
@@ -0,0 +1,68 @@
+  * DO call it a module or tab
+    * e.g. In the **Imaging Browser** module
+    * e.g. Click the **Upload** tab
+  * You can also use bold when defining features in a list, to separate the term being defined from the definition itself, e.g. when describing columns in a table:<br>
```
```suggestion
  * You can also use bold when defining features in a list to separate the term being defined from the definition itself, e.g. when describing columns in a table:<br>
```

I don't think English grammar calls for a comma here.

driusan commented on docs/HelpStyleGuide.md:
```diff
@@ -0,0 +1,68 @@
+     
+## Tone
+
+LORIS is used by scientific community members worldwide, which means it's possible that most users are non-native English speakers. Therefore, it is important that LORIS documentation (help text, setup guides, etc.) be written in a way that is quickly comprehensible by people with many different levels of English language and expression.
```
English language what and expression?

johnsaigle replied:
Maybe it would be more clear to say

```suggestion
LORIS is used by scientific community members worldwide, which means it's possible that most users are non-native English speakers. Therefore, it is important that LORIS documentation (help text, setup guides, etc.) be written in a way that is concise, clear, and free of needlessly ambiguous or complicated words.
```

driusan replied:
I like that better.. I was also a little confused about what users were expressing while reading the help text..

driusan commented on docs/HelpStyleGuide.md:
```diff
@@ -0,0 +1,68 @@
+  * e.g. navigate pages of a table using the arrows provided
+* Use Plain Language
+  * Avoid words that you wouldn't use in everyday speech
+* Where relevant, introduce an abbreviation once and then use that abbreviation for the rest of the copy in that help text (e.g. introduce Quality Control = QC in a help text title, then only call it QC throughout the text)
```
```suggestion
* Where relevant introduce an abbreviation once and then use that abbreviation for the rest of the copy in that help text (e.g. introduce Quality Control = QC in a help text title, then only call it QC throughout the text)
```

I don't think this comma is necessary either.

johnsaigle replied:
I think the comma is better than no comma. 

Maybe the "Where relevant" can be removed from the beginning of the sentence though. The sentence stands on its own without it.

Or maybe "When introducing an abbreviation, use it once and then...'"

driusan replied:
I don't feel strongly about this particular comma.

johnsaigle replied:
Nor I

christinerogers commented:
@driusan     see @knoll-alex 's latest changes and re-review? thanks!
